### ISSUE TITLE
Research `Protocol` for python as a replacement to ABC for EvalCommand

### ISSUE THREAD

**Issue by RonConfigu** [configu/configu#199]
**Description**
https://github.com/configu/configu/pull/163#discussion_r1280937192
As suggested, look into Protocol and if applicable please use it for EvalCommand

**Motivation**

**Additional context**
https://jellis18.github.io/post/2022-01-11-abc-vs-protocol/

**Comment by rannn505** [comment]
@RonConfigu can you summarize your input here? so it will be available to all.

**Comment by RonConfigu** [comment]
@rannn505 @RichardAkman @pelegpor

Hi Everyone, 
below are my thoughts regarding replacing `abc.ABC` / `typing.Generic` with `typing.Protocol` as suggested here https://github.com/configu/configu/pull/163#discussion_r1280937192

First things first, for anyone not familiar with python, anything imported from `typing` package 
**doesn't effect the program in any way**
it's only meant for developers ease of use and depending on their IDE they might see warnings.
For those with `TypeScript` as their "first" language, `Protocol == Interface`

Also taking to consideration the following:
As far as I've been instructed, the Python SDK should be **EXACTLY AS** or **as close as possible** to the Typescript SDK.

That being said, I will first address the comment's bullets and then my own opinion.
> We don't need to have parameters in the base class, allowing for parameter-less commands.

if parameter-less commands is refers to the `__init__` params then the parameters will stay as long as they're existing in TS SDK. Nothing we can change here.
if parameter-less commands refers to the class `__mro__` then it will be nice to have none-inherited class syntax although it won't change anything interpreter-wise and it will be very different from the TS SDK.

> We don't need to import subclass Command in various places

While true, to my best understanding we will need to import the protocol to define the type as the variable declaration anyway. So I don't see a big difference here.
```python
cmd_result: CmdResultProtocol = {...}
```

> Users wishing to implement their own commands don't have to subclass Command, their class just needs to implement a run method.

I'm not sure implementing user-specific commands is a requirement from Configu Product. To my best knowledge it is not. 
If it is, Protocol will not force the user to implement the `run()` and the error will raise at run-time instead of interpreter.

In my opinion, even though `Protocol` is a cool new feature which I was happy to learn about and pretty sure I could/would find usages for it. I don't see the advantage of using it considering traits, limitations and implementation/refactoring overhead.

@jamesL92 If you feel like i've missed something or didn't understand your suggestions please feel free to educate or correct me.

Thanks,
Ron
